src/main/java/org/apache/maven/plugin/resources/remote/AbstractProcessRemoteResourcesMojo.java

```diff
@@ -0,0 +1,1358 @@
+     *
+     * @since 1.0-alpha-5
+     */
+    @Parameter
+    private String[] supplementalModels;
+
+    /**
+     * List of artifacts that are added to the search path when looking
+     * for supplementalModels, expressed with <code>groupId:artifactId:version[:type[:classifier]]</code> format.
+     *
+     * @since 1.1
+     */
+    @Parameter
+    private List<String> supplementalModelArtifacts;
+
+    /**
+     * The resource bundles that will be retrieved and processed,
+     * expressed with <code>groupId:artifactId:version[:type[:classifier]]</code> format.
+     */
+    @Parameter( required = true )
+    private List<String> resourceBundles;
+
+    /**
+     * Skip remote-resource processing
+     *
+     * @since 1.0-alpha-5
+     */
+    @Parameter( property = "remoteresources.skip", defaultValue = "false" )
+    private boolean skip;
+
+    /**
+     * Attaches the resources to the main build of the project as a resource directory.
+     *
+     * @since 1.5
+     */
+    @Parameter( defaultValue = "true", property = "attachToMain" )
+    private boolean attachToMain;
+
+    /**
+     * Attaches the resources to the test build of the project as a resource directory.
+     *
+     * @since 1.5
+     */
+    @Parameter( defaultValue = "true", property = "attachToTest" )
+    private boolean attachToTest;
+
+    /**
+     * Additional properties to be passed to Velocity.
+     * Several properties are automatically added:<ul>
+     * <li><code>project</code> - the current MavenProject </li>
+     * <li><code>projects</code> - the list of dependency projects</li>
+     * <li><code>projectsSortedByOrganization</code> - the list of dependency projects sorted by organization</li>
+     * <li><code>projectTimespan</code> - the timespan of the current project (requires inceptionYear in pom)</li>
+     * <li><code>locator</code> - the ResourceManager that can be used to retrieve additional resources</li>
+     * </ul>
+     * See <a
+     * href="https://maven.apache.org/ref/current/maven-project/apidocs/org/apache/maven/project/MavenProject.html"> the
+     * javadoc for MavenProject</a> for information about the properties on the MavenProject.
+     */
+    @Parameter
+    protected Map<String, Object> properties = new HashMap<>();
+
+    /**
+     * Whether to include properties defined in the project when filtering resources.
+     *
+     * @since 1.2
+     */
+    @Parameter( defaultValue = "false" )
+    protected boolean includeProjectProperties = false;
+
+    /**
+     * When the result of velocity transformation fits in memory, it is compared with the actual contents on disk
+     * to eliminate unnecessary destination file overwrite. This improves build times since further build steps
+     * typically rely on the modification date.
+     *
+     * @since 1.6
+     */
+    @Parameter( defaultValue = "5242880" )
+    protected int velocityFilterInMemoryThreshold = 5 * 1024 * 1024;
```

Comment:
Why do you provide the default value twice?

Reply:
This is how it is in original code, did not touch this area.